src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,286 +1633,372 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
         (CHECK_WEEKEVENTREG(WEEKEVENTREG_CLEARED_SNOWHEAD_TEMPLE))) {
-        func_80A14018(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_ChangeToStretchingAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
     } else {
         Actor_Kill(&this->actor);
     }
 }
 
-void func_80A1449C(EnGo* this, PlayState* play) {
+/**
+ * Setup a Goron spectating at the racetrack.
+ *
+ * Spectators to the Goron races cannot be engaged, they simply stand idle and cheer.
+ */
+void EnGo_SetupSpectator(EnGo* this, PlayState* play) {
     if ((gSaveContext.save.entrance == ENTRANCE(GORON_RACETRACK, 1)) ||
         (gSaveContext.save.entrance == ENTRANCE(CUTSCENE, 0))) {
-        func_80A14104(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_ChangeToSpectatingAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
     } else {
         Actor_Kill(&this->actor);
     }
 }
 
-void func_80A144F4(EnGo* this, PlayState* play) {
+/**
+ * Setup the Goron Shrine's Gatekeeper.
+ *
+ * When spoken to, can open the Goron Shrine for the player.
+ * - On Day 1: Can be found in the open, shivering.
+ * - On Days 2,3: Can be found in a snowball.
+ */
+void EnGo_SetupGatekeeper(EnGo* this, PlayState* play) {
     if (gSaveContext.save.day >= 2) {
-        this->unk_284 = SubS_GetDayDependentPath(play, ENGO_GET_7F80(&this->actor), 0xFF, &this->unk_3E4);
-        if (this->unk_284 != NULL) {
-            this->unk_3E4 = 1;
+        this->gatekeeperPath = SubS_GetDayDependentPath(play, ENGO_GET_PATH(&this->actor), 0xFF, &this->indexPathPoint);
+        if (this->gatekeeperPath != NULL) {
+            this->indexPathPoint = 1;
         }
-        func_80A1428C(this, play);
-        this->actionFunc = func_80A153FC;
-        this->unk_3D8 = func_80A13B1C;
+        EnGo_ChangeToSnowballAnimation(this, play);
+        this->actionFunc = EnGo_Snowball;
+        this->msgEventFunc = EnGo_HandleOpenShrineCutscene;
     } else {
-        func_80A143A8(this, play);
-        this->actionFunc = func_80A149B0;
-        this->unk_3D8 = func_80A13B1C;
+        EnGo_ChangeToShiveringAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
+        this->msgEventFunc = EnGo_HandleOpenShrineCutscene;
     }
 }
 
-void func_80A145AC(EnGo* this, PlayState* play) {
-    if ((ENGO_GET_70(&this->actor) == ENGO_70_1) &&
-        (((play->sceneId == SCENE_10YUKIYAMANOMURA2) && (gSaveContext.sceneLayer == 1) &&
-          (play->csCtx.scriptIndex == 0)) ||
-         !CHECK_WEEKEVENTREG(WEEKEVENTREG_21_08))) {
-        this->actor.child = func_80A13400(this, play);
+/**
+ * Setup a Goron at the graveyard
+ *
+ * The Gorons are together, one shivering from the cold, another frozen solid.
+ */
+void EnGo_SetupGraveyardGoron(EnGo* this, PlayState* play) {
+    if ((ENGO_GET_SUBTYPE(&this->actor) == ENGO_GRAVEYARD_FROZEN) &&
+        (((play->sceneId == SCENE_10YUKIYAMANOMURA2) && // Snow mountain village
```

Comment:
```suggestion
        (((play->sceneId == SCENE_10YUKIYAMANOMURA2) &&
```
We try to keep these comments to the sceneId's, plus these will hopefully get english names soon